Documentation/contributing/making-changes.rst

```diff
@@ -0,0 +1,256 @@
+   |br|
+
+#. Commit the fixed files
+
+    .. code-block:: bash
+
+       $ git add my-file.c
+       $ git commit my-file.c
+       $ git push
+
+
+Submitting Your Changes to NuttX
+--------------------------------
+
+  Pull requests let you tell others about changes you've pushed to a branch in a repository on GitHub. Once a pull
+  request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits
+  before your changes are merged into the base branch.
+
+  (from GitHub's `About pull requests <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests>`_ page)
+
+Before you do a Pull Request, the NuttX team will usually want all the changes you made in your branch "squashed" into
+a single commit, so that when they review your changes, there's a clean view of the history. If there are changes
+after Pull Request review feedback, they can be separate commits. Here's the easiest way I found to do that initial
+squash before submitting the Pull Request:
+
+#. Check out my branch
+
+    .. code-block:: bash
+
+       $ git checkout my-branch
+
+#. Rename it to ``my-branch-old`` to save it, because we're going to create new clean branch to push:
+
+    .. code-block:: bash
+
+       $ git branch -m my-branch-old
+
+#. Create a new clean branch with the same name you were using before the last step:
+
+    .. code-block:: bash
+
+       $ git checkout master
+       $ git checkout -b my-branch
+
+#. Merge your saved old branch into the new one, telling git to "squash" all your commits into one (note this will
+   not commit the result; the changed files will be in your staging area, ready to be committed):
+
+    .. code-block:: bash
+
+       $ git merge --squash my-branch-old
+
+#. Commit the result
+
+    .. code-block:: bash
+
+       $ git commit
+
+#. Force-push your new clean branch to the remote— this will overwrite all your previous changes in that branch:
+
+    .. code-block:: bash
+
+       $ git push -f --set-upstream origin my-branch
+
+#. Create a GitHub Pull Request
+
+   A Pull Request is how you ask your upstream to review and merge your changes.
+
+   Here's `GitHub's instructions for creating a Pull Request <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_.
+
+   |br|
+
+#. Get Pull Request feedback and implement changes
+
+   Get suggestions for improvements from reviewers, make changes, and push them to the branch. Once the reviewers are
+   happy, they may suggest squashing and merging again to make a single commit. In this case you would repeat steps
+   1 through 6.
+
+Git Resources
+-------------
+
+* `Git Cheat Sheet (by GitHub) <https://github.github.com/training-kit/downloads/github-git-cheat-sheet.pdf>`_
+* `Git Book (online) <https://git-scm.com/book/en/v2>`_
+* `NuttX Code Contribution Workflow <https://cwiki.apache.org/confluence/display/NUTTX/Code+Contribution+Workflow>`_ –
```

Comment:
I'm not sure if we should point to the wiki here. Once the workflow is established it should be part of this documentation. For now I think the very brief "workflow" section of the docs is enough until we do so.

Reply:
@v01d I'd like to leave this here, so that people can see the full workflow that the developers are using. It's just a link for people to get more detail. Do you have objections to that?

Reply:
I'm not sure if the document in the wiki is really ready for "public consumption". If you want to leave the link, maybe add a text before like "Contribution Workflow Internal Draft:". I would like to avoid confusion for new users.

Reply:
@v01d I'll change the name to "Contribution Workflow Draft". The wiki is public on the Internet, though, so what about it isn't fit for public consumption?